include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -1383,10 +1395,28 @@ SpatialOperatorBase<dim, Number>::evaluate_velocity_divergence_term(VectorType &
 
 template<int dim, typename Number>
 void
-SpatialOperatorBase<dim, Number>::update_turbulence_model(VectorType const & velocity)
+SpatialOperatorBase<dim, Number>::update_viscosity(VectorType const & velocity) const
 {
-  // calculate turbulent viscosity locally in each cell and face quadrature point
-  turbulence_model.calculate_turbulent_viscosity(velocity);
+  AssertThrow(param.viscous_problem() and param.viscosity_is_variable(),
+              dealii::ExcMessage(
+                "Updating viscosity reasonable for variable viscosity models only."));
+
+  // reset the viscosity stored
+  // viscosity = viscosity_newtonian_limit
+  set_constant_viscosity(viscous_kernel_data.viscosity);
```

Comment:
Why does this function have an argument at all? Is the function `set_constant_viscosity()` called somewhere else in a different context?